examples/fungible-token/tests/general.rs

```diff
@@ -0,0 +1,75 @@
+use near_sdk_sim::{
+    call, deploy, init_simulator, to_yocto, view, ContractAccount, UserAccount, STORAGE_AMOUNT,
+};
+use std::str::FromStr;
+
+extern crate fungible_token;
+use fungible_token::FungibleTokenContract;
+
+near_sdk_sim::lazy_static::lazy_static! {
+    static ref TOKEN_WASM_BYTES: &'static [u8] = include_bytes!("../res/fungible_token.wasm").as_ref();
+}
+
+fn init(
+    initial_balance: u128,
+) -> (UserAccount, ContractAccount<FungibleTokenContract>, UserAccount) {
+    let master_account = init_simulator(None);
+    // uses default values for deposit and gas
+    let contract_user = deploy!(
+        // Contract Proxy
+        contract: FungibleTokenContract,
+        // Contract account id
+        contract_id: "contract",
+        // Bytes of contract
+        bytes: &TOKEN_WASM_BYTES,
+        // User deploying the contract,
+        signer_account: master_account,
+        // init method
+        init_method: new(master_account.account_id(), initial_balance.into())
+    );
+    let alice = master_account.create_user("alice".to_string(), to_yocto("100"));
```

Comment:
I really want accounts-with-a-contract and accounts-without-a-contract to be created with the same interface. They are the same thing in NEAR (even if represented by different structs in the machinery of the simulation tester), but this current interface makes them seem very different.

Maybe this:

```rust
let alice = create_account!(id: "alice".to_string(), initial_balance: to_yocto("100"), master_account: master_account);
let contract = create_account!(
    id: "contract",
    contract: &TOKEN_WASM_BYTES, // or TokenFactoryContract
    master_account: master_account,
    init_method: new(master_account.account_id(), initial_balance.into())
);
```

Or this, since the value of `deploy_and_init` in the context of simulation tests still seems pretty minimal to me, and I weigh helping people think clearly more heavily:

```rust
let alice = master_account.create_user("alice".to_string(), to_yocto("100"));
let contract = master_account.create_user("contract".to_string(), to_yocto("100"));
let contract = master_account.deploy(&TOKEN_WASM_BYTES);
call!(master_account, contract.new(master_account.account_id(), initial_balance.into());
```

(Holy smokes, this version finally made me realize that the `initial_balance` argument is being used to set the `total_supply`! Should we rename that argument?)

Or a hybrid approach:

```rust
let alice = master_account.create_user("alice".to_string(), to_yocto("100"));
let contract = master_account.create_user("contract".to_string(), to_yocto("100"));
let contract = deploy!(
    master_account: master_account,
    contract_account: contract,
    wasm_file: "../res/fungible_token.wasm", // could the bytes be loaded within the `deploy!` macro?
    init_function: new(master_account.account_id(), initial_balance.into())
);
```

(Note that this version uses `wasm_file` and `init_function` to match the `near-cli` arguments to `near deploy`. I think it makes sense to use `init_function` instead of `init_method`. If it's possible to pass just a string pointing to a file as above, then I also think it makes sense to call the argument `wasm_file` instead of `bytes` or `contract`.)

I think I lean toward the `create_account!` macro, out of all of these, with the argument name improvements from the third version. What do you think, @willemneal?

Reply:
My reservation about making account creation and deploying the same is that it differs from the CLI semantics.  Though I agree it is important to convey that contracts are just account + contract, but I think it's okay to treat them differently in this context.

Reply:
That's fair